src/main.js

```diff
@@ -7,155 +7,155 @@ Only expose a tiny API to keep internal routing safe
-          replace: true
-        });
-      }
-    };
-
-    return {
-      target: target,
-      '/': 'rewrite'
-    };
+@private
+**/
+AviatorConstructor.prototype._navigator = new Navigator();
+
+/**
+@method setRoutes
+@param {Object} routes
+**/
+AviatorConstructor.prototype.setRoutes = function (routes) {
+  this._navigator.setRoutes(routes);
+};
+
+/**
+dispatches routes to targets and sets up event handlers
+
+@method dispatch
+**/
+AviatorConstructor.prototype.dispatch = function (config) {
+  var navigator = this._navigator, navigatorConfig;
+
+  config = config || {};
+
+  navigatorConfig = {
+    pushStateEnabled: this.pushStateEnabled,
+    linkSelector:     this.linkSelector,
+    root:             this.root
+  };
+
+  // for now, just expose config limited config that we need
+  if (typeof config.inBrowser !== 'undefined' && !config.inBrowser) {
+    navigatorConfig._inBrowser = false;
+  }
+  if (typeof config.modifyUrl !== 'undefined' && !config.modifyUrl) {
+    navigatorConfig._modifyUrl = false;
   }
+  if (typeof config.silent !== 'undefined' && config.silent) {
+    navigatorConfig._silent = true;
+  }
+
+  navigator.setup(navigatorConfig);
 
+  navigator.dispatch();
+};
+
+/**
+@method navigate
+@param {String} uri to navigate to
+@param {Object} [options]
+**/
+AviatorConstructor.prototype.navigate = function (uri, options) {
+  this._navigator.navigate(uri, options);
+};
+
+/**
+ @method hrefFor
+ @param {String} uri to navigate to
+ @param {Object} [options]
+ **/
+AviatorConstructor.prototype.hrefFor = function (uri, options) {
+  return this._navigator.hrefFor(uri, options);
+};
+
+
+/**
+@method serializeQueryParams
+@param {Object} queryParams
+@return {String} queryString "?foo=bar&baz[]=boo&baz=[]oob"
+**/
+AviatorConstructor.prototype.serializeQueryParams = function (queryParams) {
+  return this._navigator.serializeQueryParams(queryParams);
+};
+
+/**
+@method getCurrentRequest
+@return {String}
+**/
+AviatorConstructor.prototype.getCurrentRequest = function () {
+  return this._navigator.getCurrentRequest();
+};
+
+/**
+@method getCurrentURI
+@return {String}
+**/
+AviatorConstructor.prototype.getCurrentURI = function () {
+  return this._navigator.getCurrentURI();
+};
+
+/**
+@method refresh
+**/
+AviatorConstructor.prototype.refresh = function () {
+  this._navigator.refresh();
+};
+
+/**
+@method rewriteRouteTo
+@param {String} newRoute
+@return {Object}
+**/
+AviatorConstructor.prototype.rewriteRouteTo = function (newRoute) {
+  var target = {
+    rewrite: function (request) {
+      this.navigate(newRoute, {
+        namedParams: request.namedParams,
+        replace: true
+      });
+    }
+  };
+
+  return {
+    target: target,
+    '/': 'rewrite'
+  };
 };
 
 if (typeof window !== 'undefined') {
-  window.Aviator = Aviator;
+  window.aviatorConstructor = new AviatorConstructor();
 }
 
-module.exports = Aviator;
+var aviatorConstructor = new AviatorConstructor();
```

Comment:
Is there a reason why we can't leave the creating of new instance up to the module that is require'ing? That way the module that is `require`'ing can create the new instance as needed